http/client/src/main/java/org/wildfly/security/http/client/ElytronHttpClient.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.http.client;
+
+import org.wildfly.security.http.client.hpi.ClientConfigProvider;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Base64;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+/**
+ * Http Client side Authentication using java 11 HttpClient
+ *
+ * @author <a href="mailto:kekumar@redhat.com">Keshav Kumar</a>
+ */
+public class ElytronHttpClient {
+
+    private static String basicAuth(String username, String password) {
+        return "Basic " + Base64.getEncoder().encodeToString((username + ":" + password).getBytes());
+    }
+    public HttpResponse<String> connect(String uri) throws Exception{
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = getRequest(uri);
+        HttpResponse<String> response =
+                client.send(request, HttpResponse.BodyHandlers.ofString());
+
+        return response;
+    }
+    public HttpRequest getRequest(String uri) throws Exception{
+        Iterator<ClientConfigProvider> serviceLoaderIterator = ServiceLoader.load(ClientConfigProvider.class).iterator();
```

Comment:
@keshav-725 the ServiceLoader is used to load implementations of the interface form classpath. This can be handy if we want to provide a possibility to extend / change the implementation of interface, we do not know who will provide the implementation or if we do not want to introduce a dependency on each provider. You can read about SPIs and ServiceLoader [here](https://docs.oracle.com/javase/tutorial/ext/basics/spi.html#the-serviceloader-class).  

For our elytron HTTP client, only we will provide an implementation of the methods getUsername, getPassword and getHttpAuthenticationType. While it might b euseful for someon to plug in their own ways to obtain credentials and auth type in the future, for now we plan for users to use Elytron's configuration with ElytronHttpClient. So there is no need to introduce service loading